### PR TITLE
Trainer class

### PR DESCRIPTION
**Pull request by wwwnev**
- Tried learning on SVHN on my machine, loss was going down.
- Setting hyper-parameters is still not fully streamlined. New issue ?
- Seed works (methodology : same losses after same amount of mini-batches while learning)

### CODE REVIEW

**Comment by Olivier-tl** [comment]
> * Setting hyper-parameters is still not fully streamlined. New issue ?

I think we should check that we have very good results ([close to what is reported for EfficientNet ](https://paperswithcode.com/sota/image-classification-on-cifar-10)) on balanced data. 
Issue #16 adds some hyperparams to the config. Not sure if there is hyperparams not addressed there.

**Review by Olivier-tl — APPROVED** [review]
